### PR TITLE
TSQL: Add indentation for SET statement

### PR DESCRIPTION
**Pull request by jpers36**
### Brief summary of the change made
Adding Indent/Dedent tags for TSQL's SET statement.

### Are there any other side effects of this change that we should be aware of?
No.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- [x] Included test cases to demonstrate any code changes, which may be one or more of the following:
  - [x] `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.

### CODE REVIEW

File: test/fixtures/rules/std_rule_cases/L003.yml [hunk 1]
```diff
@@ -643,3 +643,11 @@ test_pass_tsql_declare_indent:
...
     core:
       dialect: tsql
+
+test_pass_tsql_set_indent:
+  pass_str: |
+    SET @prv_qtr_1st_dt = CAST(@last_qtr_yr AS VARCHAR(4)) + '-' +
+        CAST(@last_qtr_first_mn AS VARCHAR(2)) + '-01'
+  configs:
+    core:
+      dialect: tsql
```
Comment (tunetheweb) [inline test/fixtures/rules/std_rule_cases/L003.yml:653]: Could we also add an example with commas?

For example (untested):

```suggestion
test_pass_tsql_set_indent:
  pass_str: |
    SET @prv_qtr_1st_dt = CAST(@last_qtr_yr AS VARCHAR(4)) + '-' +
        CAST(@last_qtr_first_mn AS VARCHAR(2)) + '-01'
  configs:
    core:
      dialect: tsql

test_pass_tsql_set_indent_multiple_params:
  pass_str: |
    SET @param1 = 1,
        @param2 = 2
  configs:
    core:
      dialect: tsql
```

Comment (tunetheweb) [reply test/fixtures/rules/std_rule_cases/L003.yml:653]: Oops. Looking TSQL (or at least SQLFluff’s understanding of TSQL) doesn’t support multiple sets! Sorry.

Comment (jpers36) [reply test/fixtures/rules/std_rule_cases/L003.yml:653]: @tunetheweb turns out that's not valid TSQL.  SET variable statements are limited to a single variable.

**Review by tunetheweb — APPROVED** [review]

**Comment by jpers36** [comment]
@tunetheweb tangentially, I did submit #2533 to add support for SET statements that set multiple session configuration settings, which *is* valid TSQL.

**Comment by tunetheweb** [comment]
> @tunetheweb tangentially, I did submit #2533 to add support for SET statements that set multiple session configuration settings, which _is_ valid TSQL.

Raised #2535 while this extra test case is in mine. Would be great to get your view on it.

**Pull request merged**